src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -145,27 +140,87 @@ internal abstract partial class AbstractProject : ForegroundThreadAffinitizedObj
             if (visualStudioWorkspaceOpt != null)
             {
                 this.EditAndContinueImplOpt = new VsENCRebuildableProjectImpl(this);
+                this.MetadataService = visualStudioWorkspaceOpt.Services.GetService<IMetadataService>();
             }
         }
 
         internal IServiceProvider ServiceProvider { get; }
-        
+
         /// <summary>
         /// Indicates whether this project is a website type.
         /// </summary>
         public bool IsWebSite { get; protected set; }
 
+        #region ObjOutputPath
+        private string _objOutputPath;
```

Comment:
I'll add comments and clean up further, sent this PR out early to know if I had broken something real bad.